sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -81,6 +86,19 @@ private[spark] class HiveExternalCatalog(client: HiveClient, hadoopConf: Configu
     withClient { getTable(db, table) }
   }
 
+  /**
+   * If the given table properties contains datasource properties, throw an exception.
+   */
+  private def verifyTableProperties(table: CatalogTable): Unit = {
+    val datasourceKeys = (table.properties.keys ++ table.storage.properties.keys)
+      .filter(_.startsWith(DATASOURCE_PREFIX))
```

Comment:
I moved it to https://github.com/apache/spark/pull/14506. If you agree with it now, I can include it in this PR